indico/modules/events/surveys/util.py

```diff
@@ -59,3 +65,44 @@ def was_survey_submitted(survey):
     if submission_id is None:
         return False
     return bool(SurveySubmission.find(id=submission_id).count())
+
+
+def generate_csv_from_survey(survey, submission_ids):
+    """Generates a CSV file from a given survey.
+
+    :param survey: `Survey` for which the user wants to export submissions
+    :param submission_ids: The list of submissions to include in the file
+    """
+    field_names = {'submitter', 'submission_date'}
+    field_names |= {'{}_{}'.format(question.title, question.id) for question in survey.questions
+                   if not isinstance(question.field, StaticTextField)}
+
+    with NamedTemporaryFile(suffix='indico.tmp', dir=Config.getInstance().getTempDir(), delete=False) as temp_file:
```

Comment:
You could use a `BytesIO` object here instead of an actual file. Then just `.seek(0)` at the end and return it. You can pass it to `send_file` instead of a file path. That way you don't need to delete it afterwards etc